src/components/MultipleChoiceProblem.tsx

```diff
@@ -0,0 +1,95 @@
+import { BaseProblemProps, NO_MORE_ATTEMPTS_MESSAGE } from './ProblemSetBlock'
+import { useCallback, useState } from 'react'
+import { Formik, Field, Form, ErrorMessage } from 'formik'
+import { mathifyElement } from '../lib/math'
+import * as Yup from 'yup'
+
+interface MultipleChoiceProps extends BaseProblemProps {
+  solutionOptions: string
+}
+
+interface MultipleChoiceFormValues {
+  response: string
+}
+
+export const MultipleChoiceProblem = ({
+  solvedCallback, exhaustedCallback, allowedRetryCallback, content, buttonText, solutionOptions,
+  correctResponse, encourageResponse, solution, retryLimit
+}: MultipleChoiceProps): JSX.Element => {
+  const [feedback, setFeedback] = useState('')
+  const [formDisabled, setFormDisabled] = useState(false)
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+  const parsedOptionValues: string[] = JSON.parse(solutionOptions)
+
+  const schema = Yup.object({
+    response: Yup.string().trim().required('Please select an answer')
+  })
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
+    }
+  }, [])
+
+  const clearFeedback = (): void => {
+    setFeedback('')
+  }
+
+  const generateOptions = (values: MultipleChoiceFormValues, isSubmitting: boolean, setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void): JSX.Element[] => {
+    const options: JSX.Element[] = []
+
+    parsedOptionValues.forEach(val => options.push(
+    <div key={val} className="form-check">
+      <label className="form-check-label">
+        <Field
+        className="form-radio-input"
```

Comment:
Based on [this documentation](https://getbootstrap.com/docs/5.2/forms/checks-radios/#radios) it looks like the correct class is `form-check-input` here (that might also explain / address the odd rendering otherwise).